instrumentation/apache-log4j-2/src/main/java/org/apache/logging/log4j/core/layout/StringBuilderEncoder_Instrumentation.java

```diff
@@ -0,0 +1,40 @@
+/*
+ *
+ *  * Copyright 2022 New Relic Corporation. All rights reserved.
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.apache.logging.log4j.core.layout;
+
+import com.newrelic.api.agent.weaver.MatchType;
+import com.newrelic.api.agent.weaver.Weave;
+import com.newrelic.api.agent.weaver.Weaver;
+
+import static com.nr.agent.instrumentation.log4j2.AgentUtil.getLinkingMetadataBlob;
+import static com.nr.agent.instrumentation.log4j2.AgentUtil.isApplicationLoggingEnabled;
+import static com.nr.agent.instrumentation.log4j2.AgentUtil.isApplicationLoggingLocalDecoratingEnabled;
+
+@Weave(originalName = "org.apache.logging.log4j.core.layout.StringBuilderEncoder", type = MatchType.BaseClass)
+public class StringBuilderEncoder_Instrumentation {
+
+    public void encode(final StringBuilder source, final ByteBufferDestination destination) {
+        // Do nothing if application_logging.enabled: false
+        if (isApplicationLoggingEnabled()) {
+            if (isApplicationLoggingLocalDecoratingEnabled()) {
+                // Append New Relic linking metadata from agent to log message
+                appendAgentMetadata(source);
```

Comment:
It looks like it would fall into this catch block on the library method that we're weaving this into. 

```
    @Override
    public void encode(final StringBuilder source, final ByteBufferDestination destination) {
        try {
            final Object[] threadLocalState = getThreadLocalState();
            final CharsetEncoder charsetEncoder = (CharsetEncoder) threadLocalState[0];
            final CharBuffer charBuffer = (CharBuffer) threadLocalState[1];
            final ByteBuffer byteBuffer = (ByteBuffer) threadLocalState[2];
            TextEncoderHelper.encodeText(charsetEncoder, charBuffer, byteBuffer, source, destination);
        } catch (final Exception ex) {
            logEncodeTextException(ex, source, destination);
            TextEncoderHelper.encodeTextFallBack(charset, source, destination);
        }
    }
```